main/src/ca/uwaterloo/flix/runtime/interpreter/Interpreter.scala

```diff
@@ -208,6 +207,40 @@ object Interpreter {
         throw InternalRuntimeException(s"Array index out of bounds: $i. Array length: ${b.elms.length}.")
       }
 
+    //
+    // NewChannel expressions.
+    //
+    case Expression.NewChannel(len, tpe, loc) =>
+      val l: Int = cast2int32(eval(len, env0, henv0, lenv0, root))
+      Value.Channel(l, tpe)
+
+    //
+    // GetChannel expressions.
+    //
+    case Expression.GetChannel(exp, tpe, loc) =>
+      val c = cast2channel(eval(exp, env0, henv0, lenv0, root))
+      c.get()
+
+    //
+    // PutChannel expressions.
+    //
+    case Expression.PutChannel(exp1, exp2, tpe, loc) =>
+      val v = eval(exp2, env0, henv0, lenv0, root)
+      val c = cast2channel(eval(exp1, env0, henv0, lenv0, root))
+      c.put(v)
+
+    //
+    // Spawn expressions.
+    //
+    case Expression.Spawn(exp, tpe, loc) =>
+      val clo = eval(exp, env0, henv0, lenv0, root)
+
+      val t = new Thread() {
```

Comment:
Looks reasonable. Remember a Thread can be given a name, might be useful for debugging.